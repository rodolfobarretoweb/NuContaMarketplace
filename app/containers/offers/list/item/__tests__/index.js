import React from 'react';
import renderer from 'react-native-test-utils';
import checkPropTypes, { assertPropTypes } from 'check-prop-types';
import { shallow } from 'enzyme';
import Item from '../index';

describe('Item component', () => {
  const defaultProps = {
    navigator: {},
    id: 'id',
    price: 10,
    balance: 1000,
    product: {
      name: 'Product name',
      description: 'Description',
      image: 'image.jpg'
    }
  };

  it('renders snapshot correctly', () => {
    const item = renderer(<Item {...defaultProps} />);
    expect(item.toJSON()).toMatchSnapshot();
  });

  it('fails without navigator prop', () => {
    expect(() => {
      assertPropTypes(Item.propTypes, { navigator: undefined }, 'prop');
    }).toThrow();
  });

  it('fails without id prop', () => {
    expect(() => {
      assertPropTypes(Item.propTypes, { id: undefined }, 'prop');
    }).toThrow();
  });

  it('fails without balance prop', () => {
    expect(() => {
      assertPropTypes(Item.propTypes, { balance: undefined }, 'prop');
    }).toThrow();
  });

  it('fails without price prop', () => {
    expect(() => {
      assertPropTypes(Item.propTypes, { price: undefined }, 'prop');
    }).toThrow();
  });

  it('fails without product name prop', () => {
    expect(() => {
      assertPropTypes(Item.propTypes, { product: { name: undefined } }, 'prop');
    }).toThrow();
  });

  it('fails without product description prop', () => {
    expect(() => {
      assertPropTypes(Item.propTypes, { product: { description: undefined } }, 'prop');
    }).toThrow();
  });

  it('fails without product image prop', () => {
    expect(() => {
      assertPropTypes(Item.propTypes, { product: { image: undefined } }, 'prop');
    }).toThrow();
  });

  it('fails with wrong prop type on `navigator` prop', () => {
    expect(
      checkPropTypes(Item.propTypes, { ...defaultProps, navigator: '' }, 'prop', 'Item')
    ).toContain('Invalid prop `navigator` of type `String` supplied to `Item`, expected instance of `Object`.');
  });

  it('fails with wrong prop type on `balance` prop', () => {
    expect(
      checkPropTypes(Item.propTypes, { ...defaultProps, balance: false }, 'prop', 'Item')
    ).toContain('Invalid prop `balance` of type `boolean` supplied to `Item`, expected `number`.');
  });

  it('fails with wrong prop type on `id` prop', () => {
    expect(
      checkPropTypes(Item.propTypes, { ...defaultProps, id: false }, 'prop', 'Item')
    ).toContain('Invalid prop `id` of type `boolean` supplied to `Item`, expected `string`.');
  });

  it('fails with wrong prop type on `price` prop', () => {
    expect(
      checkPropTypes(Item.propTypes, { ...defaultProps, price: false }, 'prop', 'Item')
    ).toContain('Invalid prop `price` of type `boolean` supplied to `Item`, expected `number`.');
  });

  it('fails with wrong prop type on `product name` prop', () => {
    expect(
      checkPropTypes(Item.propTypes, { ...defaultProps, product: { ...defaultProps.product, name: false } }, 'prop', 'Item')
    ).toContain('Invalid prop `product.name` of type `boolean` supplied to `Item`, expected `string`.');
  });

  it('fails with wrong prop type on `product description` prop', () => {
    expect(
      checkPropTypes(Item.propTypes, { ...defaultProps, product: { ...defaultProps.product, description: false } }, 'prop', 'Item')
    ).toContain('Invalid prop `product.description` of type `boolean` supplied to `Item`, expected `string`.');
  });

  it('fails with wrong prop type on `product image` prop', () => {
    expect(
      checkPropTypes(Item.propTypes, { ...defaultProps, product: { ...defaultProps.product, image: false } }, 'prop', 'Item')
    ).toContain('Invalid prop `product.image` of type `boolean` supplied to `Item`, expected `string`.');
  });

  describe('when press', () => {
    const props = {
      ...defaultProps,
      navigator: { push: jest.fn() }
    };

    const simulate = () => {
      const wrapper = shallow(<Item {...props} />);
      const render = wrapper.dive();

      render.find('TouchableHighlight').simulate('Press');

      return render;
    };

    it('redirect with push method', () => {
      simulate();
      expect(props.navigator.push).toHaveBeenCalled();
    });
  });
});
