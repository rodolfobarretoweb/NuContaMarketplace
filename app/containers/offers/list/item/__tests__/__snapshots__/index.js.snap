// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item component renders snapshot correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Object {
        "backgroundColor": "#fff",
        "elevation": 3,
        "marginBottom": 10,
        "padding": 15,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 0.1,
          "width": 1,
        },
        "shadowOpacity": 0.3,
      },
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View>
    <Image
      source={
        Object {
          "uri": "image.jpg",
        }
      }
      style={
        Object {
          "height": 150,
          "marginBottom": 10,
          "resizeMode": "cover",
          "width": "100%",
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#333",
          "fontSize": 20,
          "marginBottom": 5,
        }
      }
    >
      Product name
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#782698",
          "fontSize": 16,
          "fontWeight": "bold",
          "marginBottom": 10,
        }
      }
    >
      $
      10
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#333",
          "fontSize": 14,
        }
      }
    >
      Description
    </Text>
  </View>
</View>
`;
